src/colvarbias_restraint.cpp

```diff
@@ -766,16 +766,20 @@ int colvarbias_restraint_harmonic::update()
   // update the TI estimator (if defined)
   error_code |= colvarbias_ti::update();
 
-  // update parameters (centers or force constant)
-  error_code |= colvarbias_restraint_centers_moving::update();
-  error_code |= colvarbias_restraint_k_moving::update();
+  if (cvm::main()->proxy->simulation_running()) {
+    // update parameters (centers or force constant)
+    error_code |= colvarbias_restraint_centers_moving::update();
+    error_code |= colvarbias_restraint_k_moving::update();
+  }
```

Comment:
@jhenin For consistency I think this check should also be done for the other derived restraint classes, i.e. `colvarbias_linear` and `colvarbias_harmonic_walls`.

If so, maybe a simpler approach would be to do the check inside the two `*_moving::update()` methods in the base classes, so that derived restraint classes automatically get it. But I'm not hung up on this latter suggestion if there's a strong argument against.